ui/src/app/components/with-new-user-satisfaction-survey-modal-wrapper.tsx

```diff
@@ -0,0 +1,63 @@
+import * as React from 'react';
+import { useEffect, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+
+import { useQuery } from 'app/components/app-router';
+import { NewUserSatisfactionSurveyModal } from 'app/components/new-user-satisfaction-survey-modal';
+import { surveysApi } from 'app/services/swagger-fetch-clients';
+import { authStore, profileStore, useStore } from 'app/utils/stores';
+
+const ONE_TIME_CODE_QUERY_PARAMETER = 'surveyCode';
+
+export const withNewUserSatisfactionSurveyModal = (WrappedComponent) => {
+  return ({ ...props }) => {
+    const [showModal, setShowModal] = useState(false);
+    const { reload } = useStore(profileStore);
+    const [surveyCode, setSurveyCode] = useState(null);
+    const query = useQuery();
+    const history = useHistory();
+
+    useEffect(() => {
+      const codeParam = query.get(ONE_TIME_CODE_QUERY_PARAMETER);
+      if (codeParam == null) {
+        return;
+      }
+      setSurveyCode(codeParam);
+
+      // Deleting the query parameter prevents certain flows (such as logging out) from re-triggering a cancelled survey
+      query.delete(ONE_TIME_CODE_QUERY_PARAMETER);
```

Comment:
The one you linked is a lot more aggressive and deletes _every_ query parameter. I would prefer to only delete the relevant one in case we want to add other query parameters in the future.